Infrastructure/DataImporterService.cs

```diff
@@ -191,32 +194,228 @@ private async Task RemoveGeneralInformationAboutSchools(NtpDbContext dbContext,
+
+                dbContext.TuitionPartners.Add(tuitionPartnerToProcess);
+
+                await dbContext.SaveChangesAsync(cancellationToken);
+
+                _logger.LogInformation("Added Tuition Partner {TuitionPartnerName} with id of {TuitionPartnerId} from file {OriginalFilename}",
+                    tuitionPartnerToProcess.Name, tuitionPartnerToProcess.Id, originalFilename);
+            }
+            else if (matchedTPs.Count == 1)
+            {
+                //If matched then see if needs updating
+
+                var existingTP = matchedTPs.First();
+
+                //Map spreadsheet data to existing entity
+                existingTP = tuitionPartnerToProcess!.Adapt(existingTP);
+
+                ImportTuitionPartnerLocalAuthorityDistrictCoverage(dbContext, existingTP, tuitionPartnerToProcess);
+                ImportTuitionPartnerSubjectCoverage(dbContext, existingTP, tuitionPartnerToProcess);
+                ImportTuitionPartnerPrices(dbContext, existingTP, tuitionPartnerToProcess);
+
+                if (!existingTP.IsActive ||
+                    dbContext.ChangeTracker.Entries().Any(e => e.State == EntityState.Modified ||
+                                                                e.State == EntityState.Added ||
+                                                                e.State == EntityState.Deleted))
+                {
+                    SetImportingData(existingTP);
 
-            _logger.LogInformation("Added Tuition Partner {TuitionPartnerName} with id of {TuitionPartnerId} from file {OriginalFilename}",
-                tuitionPartner.Name, tuitionPartner.Id, originalFilename);
+                    await dbContext.SaveChangesAsync(cancellationToken);
+
+                    _logger.LogInformation("Updated Tuition Partner {TuitionPartnerName} with id of {TuitionPartnerId} from file {OriginalFilename}",
+                        existingTP.Name, existingTP.Id, originalFilename);
+                }
+                else
+                {
+                    _logger.LogInformation("No changes for Tuition Partner {TuitionPartnerName} with id of {TuitionPartnerId} from file {OriginalFilename}",
+                        existingTP.Name, existingTP.Id, originalFilename);
+                }
+            }
+            else if (matchedTPs.Count > 1)
+            {
+                var errorMsg = $"The file {originalFilename} with seo url('{tuitionPartnerToProcess.SeoUrl}') and import id('{tuitionPartnerToProcess.ImportId}') is returning more than 1 result from the database.";
+                _logger.LogError(message: errorMsg);
+                throw new InvalidOperationException(errorMsg);
+            }
+
+            successfullyProcessed.Add(originalFilename, tuitionPartnerToProcess);
         }
+
+        await DeactivateTPs(dbContext, allExistingTPs, successfullyProcessed, cancellationToken);
+    }
+
+    private static void ConfigureMapper()
+    {
```

Comment:
Can we have a `DataImporterConfig` static class that encapsulates all the mapping logic for the data importer stuff and calls it here? i.e:

```
public static class DataImporterConfig
{
    public static void Configure()
    {
        ConfigureLocalAuthorityDistrictCoverage();
        ConfigureSubjectCoverage();
        ConfigurePrice();
        ConfigureTuitionPartner();
    }

    private static void ConfigureLocalAuthorityDistrictCoverage()
    {
        TypeAdapterConfig<LocalAuthorityDistrictCoverage, LocalAuthorityDistrictCoverage>
            .NewConfig()
            .Ignore(dest => dest.Id)
            .Ignore(dest => dest.TuitionPartnerId)
            .Ignore(dest => dest.TuitionPartner!)
            .Ignore(dest => dest.TuitionType!)
            .Ignore(dest => dest.LocalAuthorityDistrict!);
    }

    private static void ConfigureSubjectCoverage()
    {
        TypeAdapterConfig<SubjectCoverage, SubjectCoverage>
            .NewConfig()
            .Ignore(dest => dest.Id)
            .Ignore(dest => dest.TuitionPartnerId)
            .Ignore(dest => dest.TuitionPartner!)
            .Ignore(dest => dest.TuitionType!)
            .Ignore(dest => dest.Subject!);
    }

    private static void ConfigurePrice()
    {
        TypeAdapterConfig<Price, Price>
            .NewConfig()
            .Ignore(dest => dest.Id)
            .Ignore(dest => dest.TuitionPartnerId)
            .Ignore(dest => dest.TuitionPartner!)
            .Ignore(dest => dest.TuitionType!)
            .Ignore(dest => dest.Subject!);
    }

    private static void ConfigureTuitionPartner()
    {
        TypeAdapterConfig<TuitionPartner, TuitionPartner>
            .NewConfig()
            .Ignore(dest => dest.Id)
            .Ignore(dest => dest.Prices!)
            .Ignore(dest => dest.LocalAuthorityDistrictCoverage!)
            .Ignore(dest => dest.SubjectCoverage!)
            .Ignore(dest => dest.Logo!)
            .Ignore(dest => dest.HasLogo)
            .Ignore(dest => dest.ImportProcessLastUpdatedData)
            .Ignore(dest => dest.IsActive)
            .Ignore(dest => dest.OrganisationType);
    }
}

```

Reply:
Done